front/app/containers/Admin/pagesAndMenu/containers/AddProjectNavbarItemModal/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from 'react';
+
+import { Box, Text } from '@citizenlab/cl2-component-library';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormProvider, useForm } from 'react-hook-form';
+import { Multiloc } from 'typings';
+import { object, string } from 'yup';
+
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+import useAddNavbarItem from 'api/navbar/useAddNavbarItem';
+import useProjects from 'api/projects/useProjects';
+
+import useLocale from 'hooks/useLocale';
+import useLocalize from 'hooks/useLocalize';
+
+import InputMultilocWithLocaleSwitcher from 'components/HookForm/InputMultilocWithLocaleSwitcher';
+import Select from 'components/HookForm/Select';
+import Button from 'components/UI/Button';
+import Modal from 'components/UI/Modal';
+import Warning from 'components/UI/Warning';
+
+import { useIntl } from 'utils/cl-intl';
+import { handleHookFormSubmissionError } from 'utils/errorUtils';
+import validateAtLeastOneLocale from 'utils/yup/validateAtLeastOneLocale';
+
+import messages from './messages';
+
+export interface FormValues {
+  projectId: string;
+  titleMultiloc: Multiloc;
+}
+
+type Props = {
+  opened: boolean;
+  onClose: () => void;
+};
+
+const AddProjectNavbarItemModal = ({ opened, onClose }: Props) => {
+  const { mutateAsync: addNavbarItem } = useAddNavbarItem();
+  const { data: projects } = useProjects({
+    publicationStatuses: ['published', 'draft', 'archived'],
+  });
+  const { data: appConfig } = useAppConfiguration();
+
+  const { formatMessage } = useIntl();
+  const localize = useLocalize();
+  const locale = useLocale();
+
+  const schema = object({
+    projectId: string().required(formatMessage(messages.emptyProjectError)),
+    titleMultiloc: validateAtLeastOneLocale(
+      formatMessage(messages.emptyNameError)
+    ),
+  });
+
+  const methods = useForm({
+    mode: 'onBlur',
```

Comment:
This defined the validation method for the form fields. It means the validation rules will trigger when the field is blurred. Another way could be `onChange` for example but we use blur because it's better UX.